capstone-ui/src/components/Search/Search.jsx

```diff
@@ -33,48 +32,52 @@ export default function Search({ username, token }) {
   };
 
   const searchSongs = async (e) => {
-    if (searchInput != "") {
-      let temp = searchInput;
-      if (searchInput === "") {
-        temp = searchInputValue;
-      }
-      setSearchInputValue(temp);
-
-      setProfileResults([]);
-
-      e.preventDefault();
-      if (!isSongResults) {
-        setOffset(0);
-      }
-
-      let oldResults = songResults;
-      const { data } = await axios
-        .get(
-          "https://api.spotify.com/v1/search",
-          {
-            params: {
-              q: temp,
-              type: "track",
-              include_external: "audio",
-              limit: 5,
-              offset: offset,
-            },
-          },
-          {
-            headers: {
-              Authorization: `Bearer ${token}`,
-            },
-          }
-        )
-        .catch((error) => {});
-
-      let newResults = data.tracks.items;
-      let addedResults = oldResults.concat(newResults);
-      setSongResults(addedResults);
-      setIsSongResults(true);
-      setOffset((previousValue) => previousValue + 5);
-      setSearchInput("");
+    let oldSearchResults;
+    let trueSearchValue = searchInput;
+    if (searchInput == "") {
+      trueSearchValue = searchInputValue;
+      oldSearchResults = songResults;
+    } else {
+      oldSearchResults = [];
     }
+
+    let postRequest = {
+      username: username,
+      searchValue: trueSearchValue,
+    };
+    const searchResponse = await axios
+      .post(`http://localhost:3001/user/addrecentsearch`, postRequest)
+      .catch((error) => alert(`Error! ${error.message}`));
+
+    const { data } = await axios
+      .get(
+        "https://api.spotify.com/v1/search",
+        {
+          params: {
+            q: trueSearchValue,
+            type: "track",
+            include_external: "audio",
+            limit: 5,
+            offset: offset,
+          },
+        },
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      )
+      .catch((error) => {
+        alert(`Error searching for ${trueSearchValue}.`);
+      });
+
+    let newResults = data.tracks.items;
+    let addedResults = oldSearchResults.concat(newResults);
```

Comment:
This can be 1 line since it doesn't seem like you actually use newResults
const addedResults = oldSearchResults.concat(data.tracks.items);

Reply:
setSongResults(oldResults.concat(data.tracks.items));
I updated this in my Search PR so that the line now is the above ^

Reply:
nice refactoring